mempool/src/filter.rs

```diff
@@ -0,0 +1,155 @@
+use account::Account;
+use collections::LimitHashSet;
+use keys::Address;
+use nimiq_hash::{Blake2bHash, Hash};
+use primitives::coin::Coin;
+use primitives::networks::NetworkId;
+use transaction::{Transaction, TransactionFlags};
+
+
+
+#[derive(Debug)]
+pub struct MempoolFilter {
+    blacklist: LimitHashSet<Blake2bHash>,
+    rules: Rules,
+}
+
+impl MempoolFilter {
+
+    pub const DEFAULT_RULES: Rules = Rules {
+        tx_fee: Coin::ZERO,
+        tx_fee_per_byte: 0.0,
+        tx_value: Coin::ZERO,
+        tx_value_total: Coin::ZERO,
+        contract_fee: Coin::ZERO,
+        contract_fee_per_byte: 0.0,
+        contract_value: Coin::ZERO,
+        creation_fee: Coin::ZERO,
+        creation_fee_per_byte: 0.0,
+        creation_value: Coin::ZERO,
+        sender_balance: Coin::ZERO,
+        recipient_balance: Coin::ZERO,
+    };
+
+    pub const DEFAULT_BLACKLIST_SIZE: usize = 25000;
+
+    pub fn new(rules: Rules, blacklist_limit: usize) -> Self {
+        let b: LimitHashSet<Blake2bHash> = LimitHashSet::new(blacklist_limit);
+        MempoolFilter {blacklist: b, rules}
+    }
+
+    pub fn blacklist(&mut self, tx: &Transaction) -> &mut Self {
+        self.blacklist.insert(tx.hash());
+        self
+    }
+
+    pub fn remove(&mut self, tx: &Transaction) -> &mut Self {
+        let hash: Blake2bHash = tx.hash();
+        self.blacklist.remove(&hash);
+        self
+    }
+
+    pub fn blacklisted(&self, tx: &Transaction) -> bool {
+        let hash: Blake2bHash = tx.hash();
+        self.blacklist.contains(&hash)
+    }
+
+    pub fn accepts_transaction(& self, tx: &Transaction) -> bool {
```

Comment:
Unnecessary space between `&` and `self`.
```suggestion
    pub fn accepts_transaction(&self, tx: &Transaction) -> bool {
```